app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -120,22 +125,20 @@ private fun collectionTabItems(collection: TabCollection) = collection.tabs.mapI
         AdapterItem.TabInCollectionItem(collection, tab, index == collection.tabs.lastIndex)
 }
 
-class SessionControlUIView(
-    container: ViewGroup,
-    actionEmitter: Observer<SessionControlAction>,
-    changesObservable: Observable<SessionControlChange>
-) :
-    UIView<SessionControlState, SessionControlAction, SessionControlChange>(
-        container,
-        actionEmitter,
-        changesObservable
-    ) {
-
-    override val view: RecyclerView = LayoutInflater.from(container.context)
+@ExperimentalCoroutinesApi
+class SessionControlView(
+    private val homeFragmentStore: HomeFragmentStore,
```

Comment:
Typically we subscribe to the store inside of the fragment and update all of the views from there. This should also tie its lifecycle to the fragment

Reply:
Just got an explanation re: this. There is some funky behavior with the store and the FragmentPreDrawManager that is needed for some of the animations. I am okay moving that into here for now 👍